src/boot/efi/pe.h

```diff
@@ -3,17 +3,26 @@
 
 #include "efi.h"
 
-EFI_STATUS pe_memory_locate_sections(
-                const void *base,
-                const char * const sections[],
-                size_t *addrs,
-                size_t *sizes);
+EFI_STATUS pe_kernel_info(const void *base, uint32_t *ret_compat_address);
+
+struct PeSectionDescriptor {
+        char name[9];
+        size_t offset;
+        size_t size;
+};
 
 EFI_STATUS pe_file_locate_sections(
                 EFI_FILE *dir,
                 const char16_t *path,
-                const char * const sections[],
-                size_t *offsets,
-                size_t *sizes);
+                struct PeSectionDescriptor **ret_sections,
+                size_t *ret_n_sections);
 
-EFI_STATUS pe_kernel_info(const void *base, uint32_t *ret_compat_address);
+EFI_STATUS pe_memory_locate_sections(
+                const void *base,
+                struct PeSectionDescriptor **ret_sections,
+                size_t *ret_n_sections);
+
+struct PeSectionDescriptor *pe_bsearch_section(
```

Comment:
that this is a binary search is an implementation detail we shouldn't really leak into the function name here, because for the caller it shouldn't really matter. Maybe just call this "pe_find_section"?

Reply:
and please do `struct PeSectionDescriptor` → `PeSectionDescriptor` everywhere (matching the typedef suggestion earlier.